### ISSUE TITLE
Microsoft OpenJDK download policies

### ISSUE THREAD

**Issue by markkenny** [autopkg/rtrouton-recipes#185]
I'm seeing an error on the download policy. Not sure why it's failing now as the latest PKG seems to have same name as before

    AutoPKG-Microsoft-OpenJDK11-ARM.jamf.recipe.yaml
        Error in com.mygroup.jamf.autopkg-microsoftopenjdk11-arm: Processor: URLTextSearcher: Error: No match found on URL: https://learn.microsoft.com/en-us/java/openjdk/download

The regex is 
(?P&lt;url&gt;https.*?microsoft-jdk-11.*?\-aarch64.pkg)

Cutting it down below gets two links. I'm not sure "(?P&lt;url&gt;" is adding 
https.*?microsoft-jdk-11.*?\-aarch64.pkg

**Comment by rtrouton** [comment]
Looks like it's failing when trying to get the version information. I'll take a look at it.

**Comment by markkenny** [comment]
My colleague offers "Shouldn't that regex be using non-greedy repetition .+?"

`(?P<url>https.+?microsoft-jdk-11.+?-aarch64.pkg)`

I love the phrase non-greedy ;-)

**Comment by NTmatter** [comment]
Doing a bit more poking around, it looks like the regex was matching the link to `${url}.sha256sum.txt`.

A slightly revised version that prevents matching a trailing dot after `.pkg`:
```regex
(?P<url>https?:.+?microsoft-jdk-11.+?-aarch64.pkg(?!\.))
```

Arbitrary-length version numbers, and macos-specific package, just in case:
```regex
(?P<url>https?:.+?microsoft-jdk-11\.[0-9\.]+?-macos-aarch64.pkg(?!\.))
```

**Comment by rtrouton** [comment]
I've now updated the Microsoft OpenJDK 11 and OpenJDK 17 recipes. The .download recipe was finding the right download link, but searching for the version information in the download URL was now failing. I found that Microsoft was putting the version information into Library/Java/JavaVirtualMachines/microsoft-11.jdk/Contents/Info.plist (for OpenJDK 11) and Library/Java/JavaVirtualMachines/microsoft-17.jdk/Contents/Info.plist (for OpenJDK 17), so I removed the version check from the download recipes and added getting the version information to the .pkg recipes.
